include/sundials/sundials_types.h

```diff
@@ -84,13 +84,18 @@ extern "C" {
  *------------------------------------------------------------------
  */
 
+#define SUN_STRING_HELPER(x) #x
+#define SUN_STRING(x)        SUN_STRING_HELPER(x)
+
 #if defined(SUNDIALS_SINGLE_PRECISION)
 
 typedef float sunrealtype;
 #define SUN_RCONST(x)     x##F
 #define SUN_BIG_REAL      FLT_MAX
 #define SUN_SMALL_REAL    FLT_MIN
 #define SUN_UNIT_ROUNDOFF FLT_EPSILON
+#define SUN_FORMAT_E      "% ." SUN_STRING(FLT_DIG) "e"
+#define SUN_FORMAT_G      "% ." SUN_STRING(FLT_DIG) "g"
```

Comment:
Looking at the updated outputs with `"% ."` I think it would be better to have this macro not include `% ` (i.e., use `#define SUN_FORMAT_G "." SUN_STRING(FLT_DIG) "g"`. This way the space after `%` can be added locally in the format string when it's needed for alignment (i.e., outputting matrix/vector entries) and in other cases (like `PrintAllStats`) the space can be excluded.

Reply:
I agree the new output file format is not ideal with double spaces for positive numbers. What about this?
```
#define SUN_FORMAT_E      "% ." SUN_STRING(FLT_DIG) "e"
#define SUN_FORMAT_G      "%." SUN_STRING(FLT_DIG) "g"
```
Note I just removed a space from the g format. We can use `SUN_FORMAT_E` for cases where alignment is needed (I think this is already the case), and `SUN_FORMAT_G` elsewhere. We could even rename them to `SUN_FORMAT_ALIGNED` and `SUN_FORMAT`. I like the simplicity and consistency of just two macros which contain the entire format specifier. Let's discuss at tomorrow's meeting.